lib/smart_name.rb

```diff
@@ -107,6 +217,31 @@ def decoded
     @decoded ||= (s.index('&') ?  HTMLEntities.new.decode(s) : s)
   end
 
+  def pretty_key
```

Comment:
This is mostly for testing and debugging.  It makes the encoded numeric parts of the key human readable and helps make sure the test cases make sense.